commit 9c655c8be9053a65886ac3e06420399a9bfdbd70
Author: Christian Lamparter <chunkeey@googlemail.com>
Date:   Sun Sep 26 23:06:56 2010 +0200

    carl9170: fix WARN_ON triggered by Broadcom HT STAs
    
    Broadcom's Windows driver for the 4313 advertises
    an ampdu density of 7 => 16 us. The AR9170 MAC on
    the other hand only supports densities up to 8 us.
    
    This patch removes the noisy WARN_ON, because
    there is nothing we can do about it.
    
    Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

--- a/drivers/net/wireless/ath/carl9170/main.c
+++ b/drivers/net/wireless/ath/carl9170/main.c
@@ -1248,7 +1248,7 @@ static int carl9170_op_ampdu_action(stru
 
 	switch (action) {
 	case IEEE80211_AMPDU_TX_START:
-		if (WARN_ON_ONCE(!sta_info->ht_sta))
+		if (!sta_info->ht_sta)
 			return -EOPNOTSUPP;
 
 		rcu_read_lock();
